### PR TITLE
browser(firefox): fix proxy auth redirect + resource redirect handling

### PR DESCRIPTION
**Pull request by aslushnikov**
This was breaking a vital invariant in our firefox network code - see
comments.

References #10095

### CODE REVIEW

File: browser_patches/firefox-beta/juggler/NetworkObserver.js [hunk 1]
```diff
@@ -133,6 +132,17 @@ class NetworkRequest {
       // use onStopRequest, but that will only happen after the last redirect has finished.
       redirectedFrom._sendOnRequestFinished();
     }
+    // In case of proxy auth, we get to requests with the same channel:
```
Comment (yury-s) [inline browser_patches/firefox-beta/juggler/NetworkObserver.js:135]: to -> two

File: browser_patches/firefox-beta/juggler/NetworkObserver.js [hunk 2]
```diff
@@ -133,6 +132,17 @@ class NetworkRequest {
...
+    // - one is pre-auth
+    // - second is with auth header.
+    //
+    // In this case, we create this NetworkRequest object with a `redirectedFrom`
+    // object, and they both share the same httpChannel.
+    //
+    // Since we want to maintain _channelToRequest map without clashes,
+    // we must call `_sendOnRequestFinished` **before** we update it with a new object
+    // here.
+    this._networkObserver._channelToRequest.set(this.httpChannel, this);
```
Comment (yury-s) [inline browser_patches/firefox-beta/juggler/NetworkObserver.js:145]: Let's add an assert that the channel is not in the map (already).

**Review by yury-s — APPROVED** [review]

File: browser_patches/firefox-beta/juggler/NetworkObserver.js (hunk 1 above)
Comment (aslushnikov) [reply browser_patches/firefox-beta/juggler/NetworkObserver.js]: doen

File: browser_patches/firefox-beta/juggler/NetworkObserver.js (hunk 2 above)
Comment (aslushnikov) [reply browser_patches/firefox-beta/juggler/NetworkObserver.js:147]: done

**Pull request merged**